chain-signatures/node/src/http_client.rs

```diff
@@ -202,9 +228,23 @@ impl MessageQueue {
                     errors.push(err);
                 } else {
                     compacted += msgs.len();
+                    let send_latency = start.elapsed().as_millis() as f64;
                     crate::metrics::SEND_ENCRYPTED_LATENCY
                         .with_label_values(&[account_id.as_str()])
-                        .observe(start.elapsed().as_millis() as f64);
+                        .observe(send_latency);
+
+                    for (id, msg_timestamp) in presignature_msg_creation_time {
+                        if let Some(client_send_delay) =
+                            crate::util::elapsed_duration_till(msg_timestamp, start)
+                        {
+                            crate::metrics::PRESIGNATURE_CLIENT_SEND_DELAY
+                                .with_label_values(&[account_id.as_str(), &id.to_string()])
+                                .observe(client_send_delay.as_secs_f64());
```

Comment:
optional, nit: The result of `with_label_values` can be stored and reused. Looking up a metric is relatively cheap but still expensive enough to have caused significant slowdown at nearcore more than once.

With the id as a label, there is not much you can do. But without it, assuming account_id is our own id and doesn't change, you can look it up once outside the loops and reuse in each iteration.